packages/core/src/saml-applications/libraries/saml-applications.ts

```diff
@@ -91,9 +101,40 @@ export const createSamlApplicationsLibrary = (queries: Queries) => {
     });
   };
 
+  const getSamlApplicationMetadataByApplicationId = async (
+    id: string
+  ): Promise<{ metadata: string; secretId: string }> => {
+    const [{ tenantId, entityId }, { id: secretId, certificate }] = await Promise.all([
+      findSamlApplicationConfigByApplicationId(id),
+      findActiveSamlApplicationSecretByApplicationId(id),
+    ]);
+
+    assertThat(entityId, 'application.saml.entity_id_required');
+
+    const tenantEndpoint = getTenantEndpoint(tenantId, EnvSet.values);
+
+    // eslint-disable-next-line new-cap
+    const idp = saml.IdentityProvider({
+      entityID: entityId,
+      signingCert: certificate,
+      singleSignOnService: [
+        {
+          Location: buildSingleSignOnUrl(tenantEndpoint, id),
+          Binding: BindingType.Redirect,
+        },
+      ],
+    });
+
+    return {
+      metadata: idp.getMetadata(),
+      secretId,
```

Comment:
Just curious why do we need this?

Reply:
We use `secretId` as part of metadata/certificate's file name.

Reply:
Removed, double checked certificates download from Azure/Okta, they are using general name like `okta.cert`.